File: problem-solving/baekjoon/4573_Pizza_Pricing.go
// https://www.acmicpc.net/problem/4573
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var menu int
	for {
		menu++
		var n int
		fmt.Fscanln(reader, &n)
		if n == 0 {
			break
		}
		var selectedDiamater int
		var lowPricePerSquare = math.MaxFloat64
		for i := 0; i < n; i++ {
			var diameter, price int
			fmt.Fscanln(reader, &diameter, &price)
			var radius = float64(diameter) / 2.0
			var pricePerSquare = float64(price) / (radius * radius * math.Phi)
			if pricePerSquare < lowPricePerSquare {
				lowPricePerSquare = pricePerSquare
				selectedDiamater = diameter
			}
		}
		fmt.Fprintf(writer, "Menu %d: %d\n", menu, selectedDiamater)
	}
}

File: problem-solving/baekjoon/4573_Pizza_Pricing.py
# https://www.acmicpc.net/problem/4573
import sys
import math

if __name__ == "__main__":
    menu = 0
    while True:
        menu += 1
        n = int(sys.stdin.readline())
        if n == 0:
            break
        selected_diamter = 0
        low_price_per_square = 10000
        for i in range(n):
            diameter, price = list(map(int, sys.stdin.readline().split()))
            radius = diameter/2.0
            price_per_square = price / (radius*radius*math.pi)
            if price_per_square < low_price_per_square:
                low_price_per_square = price_per_square
                selected_diamter = diameter
        print("Menu {}: {}".format(menu, selected_diamter))